tinygrad/codegen/linearizer.py

```diff
@@ -523,10 +523,10 @@ def ast_parse(self, x:LazyOp, acc: List[UOp], offs:Optional[List[int]], loaded_b
       assert offs is None, "not available if we aren't doing reduce"
       return acc
     # MULACC fusion. TODO: this is copied from Interpreted
-    if x.op == ReduceOps.SUM and x.src[0].__class__ is LazyOp and x.src[0].op == BinaryOps.MUL:
-      x = LazyOp(TernaryOps.MULACC, x.src[0].src, x.arg)
-    if x.op == ReduceOps.SUM and x.src[0].__class__ is LazyOp and x.src[0].op == UnaryOps.CAST and x.src[0].src[0].__class__ is LazyOp and x.src[0].src[0].op == BinaryOps.MUL:  # noqa: E501
-      x = LazyOp(TernaryOps.MULACC, x.src[0].src[0].src, x.arg)
+    if x.op == ReduceOps.SUM:
+      cast_op = x.src[0] if x.src[0].op == UnaryOps.CAST else None
+      mul_op = cast_op.src[0] if cast_op is not None and cast_op.src[0].op == BinaryOps.MUL else x.src[0] if x.src[0].op == BinaryOps.MUL else None
```

Comment:
`(cast_op.src[0] if cast_op is not None else x).src[0] == BinaryOps.MUL`

Reply:
this won't work though, what if you're just summing a localbuffer?

Reply:
there's way to rewrite the two lines to have one less if, my initially comment was incomplete and it's probably fine

Reply:
yeah i think i got it after a few tries